kratos/geometries/geometry.h

```diff
@@ -3840,15 +3840,22 @@ class Geometry
         rOStream << std::endl;
         rOStream << std::endl;
 
+        bool null_points = false;
         for (unsigned int i = 0; i < this->size(); ++i) {
             rOStream << "\tPoint " << i + 1 << "\t : ";
-            mPoints[i].PrintData(rOStream);
+            if (mPoints(i) != nullptr) {
```

Comment:
this seems like a big change

are we sure we want to allow null-points?

I think better not, imagine all the checks we would need to introduce :sweat_smile:

Reply:
We are already allowing them. All the prototype geometries use `nullptr` points.

Reply:
I mean, I don't mind dropping this, the point is that I ended up spending more time debugging this than the stuff I wanted to :smile:. IMO it makes sense to do this check for the output functions as it might be quite helpful for debugging.

Reply:
I know this was implemented a long time ago, but now we have C++17 so there's no *point* in risking segfaults because of naked raw pointers => I suggest wrapping it in an `std::optional`.

If you're worried about messing up the memory alignment, I can come up with a version of `optional` for pointers that has an identical alignment to pointers, so that's not a problem.

Reply:
I think that even though @matekelemen 's comment is a good point, it will be still good to inform developers of the null points in the `PrintData`. In other words, introducing the `std::optional` when retrieving the points doesn't exclude informing about this situation to the user.